rules/xcodeproj.bzl

```diff
@@ -785,6 +790,17 @@ Tags for configuration:
     xcodeproj-ignore-as-target: Add this to a rule declaration so that this rule will not generates a scheme for this target
 """,
     attrs = {
+        "configs": attr.string_list(mandatory = False, default = [], doc = """
+        List of bazel configs present in the .bazelrc file that can be used to build targets.
+
+        A Xcode build configuration will be created for each entry and a '--config=$CONFIGURATION' will
+        be appended to the underlying bazel invocation. Effectively allowing the configs in the .bazelrc file
+        to control how Xcode builds each build configuration.
+
+        If not present 'debug' and 'release' Xcode build configurations will be created by default without
+        appending any additional bazel invocation flags.
+        """),
+        "default_config": attr.string(mandatory = False),
```

Comment:
nit: default to `Debug`?

Reply:
Maybe i'm mistaken, but i'm not seeing the default_config getting used anywhere.

Maybe you could remove this default_config attr and just do:
```
"configs": attr.string_list(mandatory = False, default = ["Debug", "Release"], doc ...
```

Reply:
You're right! My bad, this is a leftover that shouldn't be here. Also, `configs` here should hold only configs present in a `.bazelrc` file, `["Debug", "Release"]` are created by default as Xcode build configurations only in case the user decided to not specify.

The reason for that is because in order to tell `xcodegen` bellow to create build configurations for each `configs` I have to pass the expected values to `configs` here:

```
//xcodeproj.bzl:709
    xcodeproj_info = struct(
        name = paths.split_extension(project_name)[0],
        attributes = ctx.attr.project_attributes_overrides,
        options = proj_options,
        settings = proj_settings,
        configs = xcodeproj_info_configs,
        targets = xcodeproj_targets_by_name,
        schemes = xcodeproj_schemes_by_name,
        fileGroups = project_file_groups,
    )
```
when I do that, `xcodegen` throws an error if `configs` does not receive at least of build configuration of each type `debug`/`release` (lowercased, this is the type, not the name) so `["Debug", "Release"]` receive those types and the user defined `configs` are set to type `none`.

We have two paths here:

1. The `xcodeproj` API exposes something similar to what `xcodegen` expects (making it more complex/configurable)
2. We keep the `xcodeproj` api minimal for now, and evolve it if necessary (meaning receiving just the list of names and assuming things behind the scenes)

For an initial iteration I'd suggest sticking with `2.` but I'm happy to implement it differently if folks think it's worth it atm.

Sorry for the long explanation @amberdixon I just wanted to give you the context here!

Reply:
I think you can keep the implementation code the same (which automatically sets Debug -> debug and Release -> release in the configs dictionary.) You could arguably change the defaults passed into the configs attr, so that anyone reading the rule definition will know that these defaults are gonna get applied (even if they don't know the exact bazel configuration that they map to.)